packages/core/core/src/services/document-service/transform/utils.ts

```diff
@@ -1,9 +1,72 @@
-const switchIdForDocumentId = (output: Record<string, any>) => {
+import { Common } from '@strapi/types';
+import { traverse } from '@strapi/utils';
+
+import { IdMap } from './id-map';
+
+export const switchIdForDocumentId = (output: Record<string, any>) => {
   // Mutating for performance reasons
   const documentId = output.documentId;
   delete output.documentId;
   output.id = documentId;
   return output;
 };
 
-export { switchIdForDocumentId };
+export const transformFieldsOrSort = (input: string[] | string): string[] | string => {
```

Comment:
Was thinking of ways to split this into files, rather than having it in utils.

could we have a `transform/filters.ts` `transform/sort.ts` `transform/populate.ts` ? That way we could have sepparated tests for each of them and it would be a bit easier to navigate.

wdyt?